### PR TITLE
Issues1

### PR DESCRIPTION
**Pull request by HenintsoaHARINORO**
Adding twig

### CODE REVIEW

File: test.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+$num= $_GET['vala'];
+if(isset($a)) {
+    $i = 0;
+    $res = 0;
+    for (; $i < 11; $i++) {
+        $res = $i * $a;
+        echo " $i * $a = $res <br>";
+    }
+}
+echo $twig->render('hello.html.twig', ['res' => $res] );
```
Comment (Legion112) [inline test.php]: Here you pass a variable to twig template but never use it in twig.

File: test.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+$twig = new Environment($loader, [
+    'cache' => 'cache/twig',
+]);
+$num= $_GET['vala'];
+if(isset($a)) {
+    $i = 0;
+    $res = 0;
+    for (; $i < 11; $i++) {
+        $res = $i * $a;
+        echo " $i * $a = $res <br>";
```
Comment (Legion112) [inline test.php]: What is the reason to output here?

**Review by Legion112 — CHANGES_REQUESTED** [review]

File: test.php (hunk 1 above)
Comment (HenintsoaHARINORO) [reply test.php]: I did not understand

File: test.php (hunk 2 above)
Comment (HenintsoaHARINORO) [reply test.php]: I do not know how to show the results

File: test.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?php
+require 'vendor/autoload.php';
+
+use Twig\Environment;
+use Twig\Loader\FilesystemLoader;
+
+// Initialize template engine https://twig.symfony.com/
+$loader = new FilesystemLoader('templates');
+$twig = new Environment($loader, [
+    'cache' => 'cache/twig',
```
Comment (Legion112) [inline test.php]: Remove this line twig won't use cache it will be easier to work with

**Review by HenintsoaHARINORO — COMMENTED** [review]
changes

File: test.php (hunk 2 above)
Comment (Legion112) [reply test.php]: Pass variable to twig here https://github.com/HenintsoaHARINORO/ExoPHP/pull/6/files#diff-0524b271e915e8060f768ae50abd7ea860fa5ec3b9d842218c2c207347d77f1dR21
And use those variables in the template here 
https://github.com/HenintsoaHARINORO/ExoPHP/pull/6/files#diff-2292563b834bfb6a8fdf37eb126d1e22ad8df2e239268b0f2e9148ae099e49a3R3

File: test.php [hunk 4]
```diff
@@ -9,13 +9,9 @@
...
-$num= $_GET['vala'];
-if(isset($a)) {
-    $i = 0;
-    $res = 0;
-    for (; $i < 11; $i++) {
-        $res = $i * $a;
-        echo " $i * $a = $res <br>";
-    }
+
+if(isset($_POST['vala'])) {
```
Comment (HenintsoaHARINORO) [inline test.php:10]: there is an error here,I do not know

File: test.php [hunk 5]
```diff
@@ -9,13 +9,9 @@
...
+     echo $nums=$_POST['vala'];
+
 }
-echo $twig->render('hello.html.twig', ['res' => $res] );
\ No newline at end of file
+echo $twig->render('hello.html.twig', ['num' => $nums]);
```
Comment (HenintsoaHARINORO) [inline test.php]: And here also,

**Review by HenintsoaHARINORO — COMMENTED** [review]
getting input

File: test.php [hunk 6]
```diff
@@ -6,9 +6,6 @@
 
 // Initialize template engine https://twig.symfony.com/
 $loader = new FilesystemLoader('templates');
-$twig = new Environment($loader, [
```
Comment (HenintsoaHARINORO) [inline test.php]: I removed it,and how do I return the value,I mean what should I put instead of twig variable in the return statement?

**Review by HenintsoaHARINORO — COMMENTED** [review]
$twig variables?

**Review by HenintsoaHARINORO — COMMENTED** [review]
getting from input

File: test.php (hunk 5 above)
Comment (HenintsoaHARINORO) [reply test.php]: I removed the cache and so the variable twig is gone

File: test.php (hunk 3 above)
Comment (Legion112) [reply test.php]: @HenintsoaHARINORO  by removing this line I meant lint number 10.
There is key in array named 'cache' you just need to remove `'cache' => 'cache/twig',`
 
`

File: templates/hello.html.twig [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{% extends "layout.html.twig" %}
+{% block content %}
+    <div class="column1" style="margin-top: 10px;height: 50%;margin-left: 25%;width: 50%">
+        <div class="card" style="background-color: #17a2b8">
+            <form action="multiplication.php" method="post">
```
Comment (Legion112) [inline templates/hello.html.twig]: Why `multiplication.php`?

**Review by Legion112 — CHANGES_REQUESTED** [review]

File: var/storage/messages.json [hunk 8]
```diff
@@ -0,0 +1 @@
+["dfasdfa","f","jjn","f","e"]
```
Comment (Legion112) [inline var/storage/messages.json]: Remove this file

**Review by Legion112 — CHANGES_REQUESTED** [review]

File: test.php (hunk 2 above)
Comment (HenintsoaHARINORO) [reply test.php]: The first link did not show something

File: test.php (hunk 3 above)
Comment (HenintsoaHARINORO) [reply test.php]: Ooh,OKay

File: templates/hello.html.twig (hunk 7 above)
Comment (HenintsoaHARINORO) [reply templates/hello.html.twig]: What should it be?I also did not get it

File: var/storage/messages.json (hunk 8 above)
Comment (HenintsoaHARINORO) [reply var/storage/messages.json]: okay

**Review by HenintsoaHARINORO — COMMENTED** [review]
twig variable

File: templates/hello.html.twig (hunk 7 above)
Comment (Legion112) [reply templates/hello.html.twig]: I am asking you why does this post request hanld by multiplication.php page

Comment (Legion112) [reply templates/hello.html.twig:5]: Do you know what does do `action` attribute in `form` attribute?

**Review by Legion112 — CHANGES_REQUESTED** [review]

File: test.php [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+require 'vendor/autoload.php';
+
+use Twig\Environment;
+use Twig\Loader\FilesystemLoader;
+
+// Initialize template engine https://twig.symfony.com/
+$loader = new FilesystemLoader('templates');
+$twig = new Environment($loader);
+if(isset($_POST['vala'])) {
+     echo $nums=$_POST['vala'];
```
Comment (Legion112) [inline test.php]: Explain to me what are you doing here?

File: templates/hello.html.twig [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+                Nombre:<input type="text" name="vala"> <br><br>
+                <input type="submit" value="Calculer"><br><br>
+                {% for num in nums %}
```
Comment (Legion112) [inline templates/hello.html.twig]: You have not passed variable nums to the twig template.

File: templates/hello.html.twig [hunk 11]
```diff
@@ -8,7 +8,7 @@
                 {% for num in nums %}
                     <div class="card" >
                         <div class="card-body">
-                            <p class="card-text">{{ num }}</p>
+                            <p class="card-text">{{ ress }}</p>
```
Comment (HenintsoaHARINORO) [inline templates/hello.html.twig]: I got an issue here for showing the result in the template

**Review by HenintsoaHARINORO — COMMENTED** [review]
need to fix something

File: templates/hello.html.twig (hunk 10 above)
Comment (HenintsoaHARINORO) [reply templates/hello.html.twig:8]: I have already changed it

**Review by HenintsoaHARINORO — COMMENTED** [review]
updated

**Review by HenintsoaHARINORO — COMMENTED** [review]
fixed

File: test.php [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+// Initialize template engine https://twig.symfony.com/
+$loader = new FilesystemLoader('templates');
+$twig = new Environment($loader);
+if(isset($_POST['vala'])) {
+    $nums=$_POST['vala'];
+    $i = 0;
+    $res = 0;
+    for (; $i < 11; $i++) {
+        $res = $i * $nums;
+        echo " $i * $nums = $res <br>";
```
Comment (Legion112) [inline test.php]: What does do this code?

Comment (HenintsoaHARINORO) [reply test.php]: concatenate the output,and shows it back as an arrray to hello.twig.html

**Review by HenintsoaHARINORO — COMMENTED** [review]
changes

**Review by HenintsoaHARINORO — COMMENTED** [review]
changes

File: templates/hello.html.twig [hunk 13]
```diff
@@ -7,8 +7,8 @@
                 <input type="submit" value="Calculer"><br><br>
 
 
-                {% for ress in res %}
-                    {{ ress }}
+                {% for res in ress %}
```
Comment (HenintsoaHARINORO) [inline templates/hello.html.twig]: @Legion112  here

**Review by HenintsoaHARINORO — COMMENTED** [review]
here

**Review by HenintsoaHARINORO — COMMENTED** [review]
done

**Pull request merged**